src/main/scala/io/iohk/ethereum/domain/Checkpoint.scala

```diff
@@ -0,0 +1,31 @@
+package io.iohk.ethereum.domain
+
+import io.iohk.ethereum.crypto.ECDSASignature
+import io.iohk.ethereum.rlp._
+
+case class Checkpoint(signatures: Seq[ECDSASignature])
+
+object Checkpoint {
+
+  import io.iohk.ethereum.crypto.ECDSASignatureImplicits._
+
+  implicit val checkpointRLPEncoder: RLPEncoder[Checkpoint] = _.toRLPEncodable
+
+  implicit val checkpointRLPDecoder: RLPDecoder[Checkpoint] = {
+    case RLPList(signatures: RLPList) =>
+      Checkpoint(
+        signatures.items.map(ecdsaSignatureDec.decode)
+      )
+    case _ => throw new RuntimeException("Cannot decode Checkpoint")
+  }
+
+  implicit class CheckpointEnc(checkpoint: Checkpoint) extends RLPSerializable {
+    override def toRLPEncodable: RLPEncodeable = {
+      RLPList(
+        RLPList(checkpoint.signatures.map(_.toRLPEncodable): _*)
```

Comment:
Isn't here an extra RLPList? Why the 2 RLPLists? An RLPList will already be added by the optionality of the checkpoint

Reply:
IMHO it's ok.
remember that this is an encoder for Checkpoint class.
the first RLPList is for signatures field
the second RLPList is for Checkpoint class
I know that option encoder also add RLPList, but it's for Option wrapper.

Reply:
I see what you are saying but it doesn't seem to be the way other encodings are done as well, BlockHeaders and NewBlockHashes ignore the RLPList "required" by the class the are wrapped on and are just an rlplist of headers/hashes

It also depends on how you think of this:
1. as a helper that's just an analogous of having the header field as `checkpoint: Option[Seq[Signature]] `
2. as an integral part of the header structure. This would tend to be as you implemented, but it's not followed on the examples I mentioned

Given all that I'm more inclined to remove one of the RLPList